apps/wallet-mobile/src/Nfts/filterNfts.ts

```diff
@@ -6,3 +7,16 @@ export const filterNfts = (searchTerm: string, nfts: TokenInfo[]): TokenInfo[] =
     searchTermLowerCase.length > 0 ? nfts.filter((nft) => nft.name?.toLowerCase().includes(searchTermLowerCase)) : nfts
   return filteredNfts
 }
+
+let timeoutId: ReturnType<typeof setTimeout>
+let lastSearch = ''
+export const filterNftsMetrics = (searchTerm: string, resultCount: number) => {
```

Comment:
My initial though would be that useEffects tend to rerun more than necessary.
And that those variables are not global. They are scoped to the file, which modern javascript treats as an independent module. So they can't be accessed anywhere else but that file.

I'm gonna test it but I think it will either run more than necessary, or the timeout won't clear when it should.

Reply:
Tested. It doesn't rerun for no reason, which on thinking now makes sense since string and number don't have the same issues as new objects that the other useEffect had.

But the debounce doesn't work. It calls every keystroke. You need the timeoutId outside of the function, otherwise it gets created every time and the clear is called on unmount, which doesn't really happen.